resources/packs/core/core.utils.go

```diff
@@ -35,22 +35,12 @@ func ToBool(b *bool) bool {
 	return *b
 }
 
-func StrSliceToInterface(value []string) []interface{} {
-	res := make([]interface{}, len(value))
-	for i := range value {
-		res[i] = value[i]
+func SliceToInterfaceSlice[T any](ss []T) []interface{} {
```

Comment:
Can we add a simple test case that checks:

- slice of strings to interface
- slide of strings with pointers and nil elements (this has worked before and nil elements where filtered, the new implementation behaves slightly different)

Reply:
I added tests. Also made sure the behaviour remains the same although I don't think it's needed. The ptr function was used only in 1 place for azure sql resources. I don't think they return nil entries in slices. It could be that this check was added during development for some testing. Anyways, now the behaviour is identical

Reply:
If it is just one place, lets test it without the old behavior and simplify the implementation

Reply:
If we want this to be a generic function, I think we should keep the nil checks in place. The signature of that function allows you to pass in an array with nils in it, so we should assume that this can happen, otherwise we're making assumptions about the consumers of that function. 

Right now it's used only in the azure stuff which *probably* will never give you back a nil pointer amongst the other stuff, but maybe 20 new API providers down the road some of those have weird APIs that also give you back nil entries in a list of pointers.

Instead, if you don't wanna keep that generic function then we can get rid of it and just do that conversion in the azure-related code directly